### ISSUE TITLE
latexmk / latexmkrc

### ISSUE THREAD

**Issue by inktvis75** [michal-h21/tex4ebook#61]
Hi,
Is it possible, to compile the tex file to epub, using a latexmkrc file. My latexmkrc: 

$clean_ext = "log out latexmk gz aux fls snm nav bbl *.vrb";
@default_files = ('itg203-slides.tex','itg203-labs.tex');
$pdf_mode = 4;
$pdf_update_method = 4;
$out_dir = 'build';
$postscript_mode = 0;
$dvi_mode = 0;
$lualatex = 'lualatex %O -shell-escape %S';

The problem is the out_dir, a lot of minted files are created there, that couldn't be found by the compile procedure of tex4ebook.

(Error: cannot read infile: [Errno 2] No such file or directory: 'build/itg203-labs.pyg')

**Comment by michal-h21** [comment]
You can try the `latexmk_build` extension to execute Latexmk. It just needs the development version of `tex4ebook`, as I just fixed a bug that prevented some extensions to execute. You can execute it like:

       tex4ebook -f epub3+latexmk_build -m draft -s sample.tex

You can find more info about extensions in the [make4ht doc](http://mirrors.ctan.org/support/make4ht/make4ht-doc.pdf) as they are provided by `make4ht`. Your sample `latexmkrc` file throws some error messages though. But compilation seems to finish correctly, so it is probably nothing serious.

**Comment by inktvis75** [comment]
Hi, thank you for your fast reply. I cloned the repo, installed it. 
Error message:
exec_epub.lua:73: attempt to index a nil value (field 'config_file')

(running it on macos)

**Comment by michal-h21** [comment]
Ah, it seems that it uses wrong version of the `tex4ebook` command. What path does command `which tex4ebook` says?

**Comment by inktvis75** [comment]
you're right, there are two versions available now, of course on the filesystem.
Checked the 'make install' again, and...
so, executed /Users/frederik/Library/texmf/tex/latex/tex4ebook now :)
Unfortunaly, it does not work, it does not use latexmk. From the first lines of the output:

dvilualatex itg203-labs "xhtml,,charset=utf-8,epub3,html5,uni-html4" " -cmozhtf -utf8" "" "-shell-escape -jobname=itg203-labs "

**Comment by michal-h21** [comment]
It should say some info at the beginning of the command line output. Something like:

```
Extension       +latexmk_build
Load extension  latexmk_build
```

It the loading of the extension fails, it should be:

```
Extension       +latexmk_build
Cannot load extension: latexmk_build
```

You may need to update the make4ht as well. Or wait few days, I plan to update CTAN with new versions of both `tex4ebook` and `make4ht` soon. The problem is that I keep finding new bugs every day, so I am not sure how soon it will be.

**Comment by inktvis75** [comment]
Extension	+latexmk_build
Load extension	latexmk_build

So that's ok. I do think that it's possible that it has something to do with conflicting parameters:

```
if args.lua then
latex_cmd="dvilualatex"
 elseif args.xetex then
 latex_cmd="xelatex --no-pdf"
end
```
And, if i don't add the -l parameter, it's defaulting to pdflatex, so it's ignoring the latexmkrc file.
Tried .tex4ebook, parameters local latex_cmd and latex_par seems to be ignored as well

**Comment by inktvis75** [comment]
in mkutils.lua
``
-- local newcommand = 'latexmk -latex="${expanded}" -dvi ${tex_file}' % par
  local newcommand = 'latexmk' % par
``

fixes it. Actually, I do think that if someone is going to use latexmk, he wants his own configuration file.

**Comment by michal-h21** [comment]
It is necessary to pass the `-latex` option to Latexmk, in order to load `tex4ht`. The conversion cannot be executed otherwise. Maybe it would be possible to introduce some conditional statements in the `latexmkrc` to select variables depending on the compilation tool?

**Comment by inktvis75** [comment]
ok, back to basics, i can live without latexmk, just using lualatex/dvilualtex is fine, we can try another route ... but then we're back with my original problem :)

``lualatex --shell-escape --output-directory=build itg203-labs --jobname=itg203-labs ``

is ok for me, if a dvi is needed, --output-format=dvi can be added
but, the -d parameter of tex4ebook doesn't respect the output-directory parameter

**Comment by michal-h21** [comment]
Ah, I still don't understand :(

      lualatex --shell-escape --output-directory=build filename.tex 

throws error for me, same as when I pass the `--output-directory` to `tex4ebook` LaTeX parameters. The `-d` parameter of `tex4ebook` doesn't care about intermediate files at all, it just moves the generated output file to the destination directory.
